test/unit/rules/no-heading-inside-button-test.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/no-heading-inside-button').ERROR_MESSAGE;
+
+generateRuleTests({
+  name: 'no-heading-inside-button',
+
+  config: true,
+
+  good: [
+    '<button></button>',
+    '<button><span></span></button>',
+    '<button><div></div></button>',
+    '<div></div>',
```

Comment:
Can you add `<div><h1></h1></div>` as a valid test case to make sure headings are allowed inside other elements?